keyboard.js

```diff
@@ -0,0 +1,177 @@
+        };
+
+        keyLayout.forEach(key => {
+            const keyElement = document.createElement('button');
+            const insertLineBreak = ['backspace', 'p', 'enter', '?'].indexOf(key) !== -1;
+
+            // Add attributes/classes
+            keyElement.setAttribute('type', 'button');
+            keyElement.classList.add('keyboard__key');
+
+            switch (key) {
+
+                case 'backspace':
+                    keyElement.classList.add('keyboard__key--wide');
+                    keyElement.innerHTML = createIconsHTML("backspace");
+
+                    keyElement.addEventListener('click', () => {
+                        this.properties.value = this.properties.value.substring(0, this.properties.value.length - 1);
+                        this._triggerEvent('oninput');
+                    });
+                    break;
+
+                case 'caps':
+                    keyElement.classList.add('keyboard__key--wide', 'keyboard__key--activatable');
+                    keyElement.innerHTML = createIconsHTML('keyboard_capslock');
+
+                    keyElement.addEventListener('click', () => {
+                        this._toggleCapsLock();
+                        keyElement.classList.toggle('keyboard__key--active', this.properties.capslock);
+                    });
+                    break;
+
+                case 'enter':
+                    keyElement.classList.add('keyboard__key--wide');
+                    keyElement.innerHTML = createIconsHTML('keyboard_return');
+
+                    keyElement.addEventListener('click', () => {
+                        this.properties.value += '\n';
+                        this._triggerEvent('oninput');
+                    });
+                    break;
+
+                case 'space':
+                    keyElement.classList.add('keyboard__key--extra-wide');
+                    keyElement.innerHTML = createIconsHTML('space_bar');
+
+                    keyElement.addEventListener('click', () => {
+                        this.properties.value += ' ';
+                        this._triggerEvent('oninput');
+                    });
+                    break;
+
+                case 'done':
+                    keyElement.classList.add('keyboard__key--wide');
+                    keyElement.innerHTML = createIconsHTML('check_circle', 'keyboard__key--dark');
+
+                    keyElement.addEventListener('click', () => {
+                        this.properties.value += ' ';
+                        this.close();
+                        this._triggerEvent('onclose');
+                    });
+                    break;
+
+                default:
+                    keyElement.textContent = key.toLowerCase();
+
+                    keyElement.addEventListener('click', () => {
+                        this.properties.value += this.properties.capsLock ? key.toUpperCase() : key.toLowerCase();
```

Comment:
it doesn't save values that you entered from usual keyboard. In all other places either

Reply:
fixed in line 124